buildspecs/pr-build-amzn.yml

```diff
@@ -0,0 +1,120 @@
+version: 0.2
+
+env:
+  variables:
+    # Github username of the forked repo on which to make builds
+    GITHUBUSERNAME: aws
+
+phases:
+  install:
+    commands:
+      # Same buildspec used for different Amazon Linuc versions - detect verison to use in Go installation
+      - |
+        if [ -f /etc/os-release ]; then
+          . /etc/os-release
+          case $ID:$VERSION_ID in
+            amzn:2)
+              amzn_version="amzn2"
+              ;;
+            amzn:2023)
+              amzn_version="amzn2023"
+              ;;
+            *)
+              echo "Unsupported Linux distribution: $ID:$VERSION_ID"
+              exit 1
+              ;;
+          esac
+        else
+          echo "Unable to detect the Linux distribution"
+          exit 1
+        fi
+
+      # Same buildspec for different architectures - detect the architecture here and rename the artifacts accordingly
+      - architecture=""
+      - case $(uname -m) in
+           x86_64)
+             architecture="amd64"
+           ;;
+           aarch64)
+             architecture="arm64"
+           ;;
+        esac
+
+      # Identify the correct AL2023 version to use in the Go installation
+      - AL23_VERSION="$(curl -s https://al2023-repos-us-west-2-de612dc2.s3.dualstack.us-west-2.amazonaws.com/core/releasemd.xml | xmllint --xpath "string(//root/releases/release[last()]/@version)" -)"
+      - AGENT_VERSION=$(cat VERSION)
+
+      # Need to install GOLANG explicitly as required versions do not come preinstalled
+      # Remove existing go installation (goenv utility) and install downloaded binaries
+      - sudo rm -rf /root/.goenv
+      - sudo rm -rf /usr/local/go/bin/go 
+
+      # Install dependencies and define variables based on the build image used
+      # Amazon Linux 2023 uses package manager DNF, while older versions use YUM
+      # Set the appropriate AL23 version echo string to include in build log
+      - |
+        if [[ "$amzn_version" = "amzn2023" ]]; then
+          sudo dnf --releasever="$AL23_VERSION" update -y
+          sudo yum install -y golang
```

Comment:
should we install a specific go version? especially the one in https://github.com/aws/amazon-ecs-agent/blob/master/GO_VERSION.

Reply:
Good point. Installing go needs to be handled differently for Amazon Linux, but we should still be specifying the version based on the GO_VERSION file. Currently testing the update with the CodeBuild projects now and will update this PR accordingly after. Thank you.

Reply:
Should we actually install the latest version of golang that Amazon Linux has available? The reason being that this is what Amazon Linux will be doing when it builds ecs-agent in their koji build servers, so we might actually want to mimic that behavior in our codebuild jobs.

Reply:
thats a good point. I noticed the other workflows install a specific Go version: https://github.com/aws/amazon-ecs-agent/blob/86d635bcde50bd95a17c8b59db67cfdca20badff/buildspecs/pr-build.yml#L23. 

I think we should align all pr workflow files irrespective of what we end up doing, I am ok with either options.

Reply:
Oh I understand what you are saying. Do you have any resource that I could check how the Koji builds install Go? The originally installation I had was installing the latest version of Go, but not necessarily the latest version available on Amazon Linux. [AL23](https://docs.aws.amazon.com/linux/al2023/ug/go.html), for example, does update Go with its releases, but will not always be in "lockstep" with the upstream Go releases. 

Perhaps checking for the latest available version of Go for each AL version would be best. For example:
`# Define the latest Go version available in Amazon Linux package repositories
- GOVERSION=$(yum list golang --showduplicates | awk -F'.' '{print $1"."$2}' | sort -r | head -1)`

As the team and Agent build veterans, which of these would be best? 

1. Install the latest Go version (original code)
2. Install the Go version defined in the GO_VERSION file
3. Install the latest Go version available in the AL releases. 

The answer likely lies in how Koji build servers handle Go. I will keep digging, but would value your input.